src/Strategy/Python/Poetry.hs

```diff
@@ -119,7 +124,10 @@ buildGraphWithLock lockProject poetryProject = promoteToDirect isDirect graph
     isDirect :: Dependency -> Bool
     isDirect dep = case pyprojectPoetry poetryProject of
       Nothing -> False
-      Just _ -> any (\n -> dependencyName n == (dependencyName dep)) (getDependencies poetryProject)
+      Just _ -> any (\n -> dependencyName n == dependencyName dep) (pyProjectDeps poetryProject)
+
+    lowerCasedPkgName :: PackageName -> PackageName
+    lowerCasedPkgName name = PackageName <$> toLower $ unPackageName name
```

Comment:
`toLower` is a pure function, there's no reason to use the `fmap` operator here.

It works because `-> a` is a functor, which is crazy, but necessary for something, probably.  I don't really know.
You can see it [in the source for the base library](https://hackage.haskell.org/package/base-4.15.0.0/docs/src/GHC-Base.html#line-983)

We usually don't want to use that instance, and should prefer to use `$` or `.`

I got caught by this one recently as a typo, or just not realizing that I wasn't using a "real" functor.

```suggestion
    lowerCasedPkgName name = PackageName . toLower $ unPackageName name
```